diff --git bob/io/image/test.py bob/io/image/test.py
index 5c0cebc..66471d7 100644
--- bob/io/image/test.py
+++ bob/io/image/test.py
@@ -85,7 +85,12 @@ def transcode(filename):
     assert not subsample.flags.contiguous
     write(subsample, tmpnam_)
     image3 = load(tmpnam_)
-    assert numpy.array_equal(subsample, image3)
+    try:
+      assert numpy.array_equal(subsample, image3)
+    except AssertionError:
+      print(subsample)
+      print(image3)
+      raise
 
   finally:
     if os.path.exists(tmpname):

